orientjs/orientjs.d.ts

```diff
@@ -0,0 +1,500 @@
+            set(param: string[]): Statement;
+            set(param: any): Statement;
+            content(param: string): Statement;
+            content(param: any): Statement;
+            increment(property: string, value: any): Statement;
+            add(property: string, value: any): Statement;
+            remove(property: string, value: any): Statement;
+            put(property: string, keysValues: any): Statement;
+            upsert(condition?: string, params?: any, comparisonOperator?: string): Statement;
+            upsert(condition?: any, params?: any, comparisonOperator?: string): Statement;
+            where(param: string): Statement;
+            where(param: string[]): Statement;
+            where(params: any): Statement;
+            while(param: string): Statement;
+            while(param: string[]): Statement;
+            while(param: any): Statement;
+            
+            containsText(param: any): Statement;
+            and(param: string): Statement;
+            and(param: string[]): Statement;
+            and(param: any): Statement;
+            or(param: string): Statement;
+            or(param: string[]): Statement;
+            or(param: any): Statement;
+            group(param: string): Statement;
+            group(param: string[]): Statement;
+            group(param: any): Statement;
+            order(param: string): Statement;
+            order(param: string[]): Statement;
+            order(param: any): Statement;
+            skip(value: number): Statement;
+            offset(value: number): Statement;
+            limit(value: number): Statement;
+            fetch(param: string): Statement;
+            fetch(param: any): Statement;
+            let(name: string, value: string): Statement;
+            let(name: string, value: Statement): Statement;
+            lock(param: string): Statement;
+            lock(param: any): Statement;
+            commit(retryLimit?: number): Statement;
+            retry(retryLimit?: number): Statement;
+            wait(waitLimit: number): Statement;
+            return(value: string): Statement;
+            lucene(property: string, luceneQuery: string): Statement;
+            lucene(property: any, luceneQuery: string): Statement;
+            near(latitudeProperty: string, longitudeProperty: string, longitude: number, latitude: number, maxDistanceInKms: number): Statement;
+            near(prop: any, maxDistanceInKms: number, longitude: number, latitude: number): Statement;
+
+            within(latitudeProperty: string, longitudeProperty: string, box: Array<Number>): Statement;
+
+            addParams(key: string, value: any): Statement;
+            addParams(value: any): Statement;
+            token(value: any): Statement;
+            buildStatement(): string;
+
+        }
+
+        interface Query<T> {
+
+            transform<T>(transformer: any): Query<T>;
+            transform<T>(transformer: Function): Query<T>;
```

Comment:
this signature will never be matched as overload resolution in TS goes in order of signatures, and the previous signature is general enough to match any valid call. so similar to my previous comments about `any`, do not include the last signature, it is not needed.